backend/app.ts

```diff
@@ -2,38 +2,60 @@ import * as dotenv from 'dotenv';
 import express from 'express';
 import cors from 'cors';
 import morgan from 'morgan';
-import session from 'express-session';
 import passport from 'passport';
 import googleOAuthRouter from './routes/googleOAuth';
 import testRouter from './routes/testRoute'
 import prisma from './utils/prismaClient';
-import loginAuth from './routes/loginAuth'
 import googleStrategy from 'passport-google-oauth20';
+import expressSession from 'express-session';
+import { PrismaSessionStore } from '@quixo3/prisma-session-store';
+import { PrismaClient } from '@prisma/client';
+import facebookStrategy from 'passport-facebook';
+import facebookOAuthRouter from './routes/facebookOAuth';
+import loginAuth from './routes/loginAuth'
 import { Strategy as LocalStrategy } from 'passport-local';
 import crypto from 'crypto';
+import session from 'express-session';
+import { secretcode, googleClientID, googleClientSecret, facebookAppSecret, facebookClientID } from './utils/config'
+import { request } from 'http';
+
 dotenv.config();
 const GoogleStrategy = googleStrategy.Strategy;
+const FacebookStrategy = facebookStrategy.Strategy;
 
-const db = prisma;
 const app = express();
+dotenv.config();
+const db = prisma;
 
 app.use(morgan("dev"));
 app.use(cors());
 app.use(express.json());
 
-const secretcode = process.env.SESSION_SECRET as string;
-const clientID = process.env.GOOGLE_CLIENT_ID as string;
-const clientSecret = process.env.GOOGLE_CLIENT_SECRET as string;
+app.use(
+    expressSession({
+        cookie: {
+            maxAge:  3 * 24 * 60 * 60 * 1000 //3 days
+        },
+        secret: secretcode,
+        resave: false,
+        saveUninitialized: false,
+        store: new PrismaSessionStore(
+            new PrismaClient(),
+            {
+                checkPeriod: 2 * 60 * 1000, //2 minutes
+                dbRecordIdIsSessionId: true,
+                dbRecordIdFunction: undefined,
```

Comment:
should this be undefined? if this is undefined, do we need to keep it here?

Reply:
dbRecordIdFunction: A function to generate the Prisma Record ID for a given session ID.

I just kept it there just to be explicit.